### PR TITLE
Adding Keyboards from Krado Industries

### PR DESCRIPTION
**Pull request by Feags**
## Description

Adding three keyboards designed by Krado Industries: Kousa TKL, Promenade, and Krado66. All three keyboards will have a via keymap in addition to default keymap.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kradoindustries/kousa/config.h [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+/* key matrix size */
+#define MATRIX_ROWS 6
+#define MATRIX_COLS 17
+
```
Comment (waffle87) [inline keyboards/kradoindustries/kousa/config.h:23]:
```suggestion
```

File: keyboards/kradoindustries/kousa/config.h [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+#define RGBLIGHT_EFFECT_RGB_TEST
+#define RGBLIGHT_EFFECT_ALTERNATING
+#define RGBLIGHT_EFFECT_TWINKLE
+
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+
+/* define if matrix has ghost (lacks anti-ghosting diodes) */
+
```
Comment (waffle87) [inline keyboards/kradoindustries/kousa/config.h:67]:
```suggestion
```

File: keyboards/kradoindustries/kousa/info.json [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    },
+
+    "rgblight": {
+        "led_count": 18,
+        "pin": "GP4",
+        "hue_steps": 10,
+        "saturation_steps": 17,
+        "brightness_steps": 17,
+        "animations": {
+            "all": true,
```
Comment (waffle87) [inline keyboards/kradoindustries/kousa/info.json:29]:
```suggestion
        "brightness_steps": 17,
        "sleep": true,
        "animations": {
            "alternating": true,
            "breathing": true,
            "christmas": true,
            "knight": true,
            "rainbow_mood": true,
            "rainbow_swirl": true,
            "rgb_test": true,
            "snake": true,
            "static_gradient": true,
            "twinkle": true
```

File: keyboards/kradoindustries/kousa/info.json [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+                {"matrix": [5, 14], "x":15.25, "y":5.25}, 
+                {"matrix": [5, 15], "x":16.25, "y":5.25}, 
+                {"matrix": [5, 16], "x":17.25, "y":5.25}
+            ]
+        }
+    }
+}
+// Replace label with matrix formula 
+// Find "label"\:"(\d+),(\d+)"
+// Replace "matrix": \[\1, \2\]
```
Comment (waffle87) [inline keyboards/kradoindustries/kousa/info.json:145]:
```suggestion
```

File: keyboards/kradoindustries/kousa/keymaps/default/readme.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# The default keymap for Kousa TKL
```
Comment (waffle87) [inline keyboards/kradoindustries/kousa/keymaps/default/readme.md:1]: This file should be removed.

File: keyboards/kradoindustries/kousa/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+  // First encoder (E1)
+  if (index == 0) {
+    if (clockwise) {
+      tap_code(KC_PGDN);
+    } else {
+      tap_code(KC_PGUP);
+    }
+  }
+    return true;
```
Comment (waffle87) [inline keyboards/kradoindustries/kousa/keymaps/default/keymap.c:125]:
```suggestion
    return false;
```

File: keyboards/kradoindustries/kousa/keymaps/via/readme.md [hunk 7]
```diff
@@ -0,0 +1 @@
+# Via keymap for VIA
```
Comment (waffle87) [inline keyboards/kradoindustries/kousa/keymaps/via/readme.md:1]: This file should be removed.

File: keyboards/kradoindustries/kousa/kousa.h [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
```
Comment (waffle87) [inline keyboards/kradoindustries/kousa/kousa.h:17]: This file should be removed.

File: keyboards/kradoindustries/kousa/rules.mk [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+
+# Build Options
+#   change yes to no to disable
+#
+#BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+#MOUSEKEY_ENABLE = yes       # Mouse keys
+#EXTRAKEY_ENABLE = yes       # Audio control and System control
+#CONSOLE_ENABLE = no         # Console for debug
+#COMMAND_ENABLE = yes        # Commands for debug and configuration
+#NKRO_ENABLE = yes           # Enable N-Key Rollover
```
Comment (waffle87) [inline keyboards/kradoindustries/kousa/rules.mk:15]:
```suggestion
```

File: keyboards/kradoindustries/kousa/info.json [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+    },
+    "features": {
+        "bootmagic": true,
+        "mousekey": true,
+        "extrakey": true,
+        "console": false,
+        "command": false,
+        "nkro": true
+    },
+
```
Comment (waffle87) [inline keyboards/kradoindustries/kousa/info.json:21]:
```suggestion
    "encoder": {
        "rotary": [
            { "pin_a": "GP10", "pin_b": "GP9" }
        ]
    },
```

File: keyboards/kradoindustries/krado66/config.h [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
```
Comment (waffle87) [inline keyboards/kradoindustries/krado66/config.h:19]: This file should be removed.

File: .vscode/settings.json [hunk 12]
```diff
@@ -27,5 +27,18 @@
     "python.formatting.provider": "yapf",
     "[json]": {
 	    "editor.formatOnSave": false
-    }
+    },
```
Comment (drashna) [inline .vscode/settings.json:30]: please revert.

File: c_cpp_properties.json
Comment (drashna) [inline c_cpp_properties.json:1]: please remove.

File: keyboards/kradoindustries/krado66/info.json [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+        "pin": "B2",
+        "hue_steps": 12,
+        "sleep": false,
+        "saturation_steps": 15,
+        "brightness_steps": 12,
+        "animations": {
+            "all": true,
+        }
+    },
+
```
Comment (waffle87) [inline keyboards/kradoindustries/krado66/info.json:33]:
```suggestion
    "encoder": {
        "rotary": [
            { "pin_a": "B1", "pin_b": "B0" },
            { "pin_a": "D7", "pin_b": "D6" },
            { "pin_a": "F7", "pin_b": "F6" }            
        ]
    },
```

File: keyboards/kradoindustries/krado66/info.json [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+    "rgblight": {
+        "led_count": 14,
+        "pin": "B2",
+        "hue_steps": 12,
+        "sleep": false,
+        "saturation_steps": 15,
+        "brightness_steps": 12,
+        "animations": {
+            "all": true,
```
Comment (waffle87) [inline keyboards/kradoindustries/krado66/info.json:30]:
```suggestion
        "brightness_steps": 12,
        "sleep": true,
        "animations": {
            "alternating": true,
            "breathing": true,
            "christmas": true,
            "knight": true,
            "rainbow_mood": true,
            "rainbow_swirl": true,
            "rgb_test": true,
            "snake": true,
            "static_gradient": true,
            "twinkle": true
```

File: keyboards/kradoindustries/krado66/info.json [hunk 15]
```diff
@@ -0,0 +1,118 @@
...
+                {"matrix": [9, 6], "x":13, "y":4}, 
+                {"matrix": [8, 7], "x":14, "y":4}, 
+                {"matrix": [9, 7], "x":15, "y":4}
+            ]
+        }
+    }
+}
+// Replace label with matrix formula 
+// Find "label"\:"(\d+),(\d+)"
+// Replace "matrix": \[\1, \2\]
```
Comment (waffle87) [inline keyboards/kradoindustries/krado66/info.json:118]:
```suggestion
```

File: keyboards/kradoindustries/krado66/krado66.h [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+ */
+
+#pragma once
+
+#include "quantum.h"
+
```
Comment (waffle87) [inline keyboards/kradoindustries/krado66/krado66.h:21]: This file should be removed.

File: keyboards/kradoindustries/kousa/config.h (hunk 1 above)
Comment (drashna) [inline keyboards/kradoindustries/kousa/config.h:23]: This shouldn't be needed, as it's calculated and generated from the pin assignments
```suggestion
```

File: keyboards/kradoindustries/kousa/info.json (hunk 3 above)
Comment (drashna) [inline keyboards/kradoindustries/kousa/info.json:29]: Could you explicitly define which animations are enabled? 
```suggestion
            "alternating": true,
            "breathing": true,
            "christmas": true,
            "knight": true,
            "rainbow_mood": true,
            "rainbow_swirl": true,
            "rgb_test": true,
            "snake": true,
            "static_gradient": true,
            "twinkle": true
```

File: keyboards/kradoindustries/krado66/rules.mk [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+#
+
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+RGBLIGHT_ENABLE = yes  # Enable WS2812 RGB underlight.  Do not enable this with audio at the same time.
+# RGB_MATRIX_DRIVER = WS2812
+ENCODER_ENABLE = yes
+RGBLIGHT_ENABLE = yes
+UNICODE_ENABLE = yes        # Unicode
```
Comment (waffle87) [inline keyboards/kradoindustries/krado66/rules.mk:42]:
```suggestion
ENCODER_ENABLE = yes
RGBLIGHT_ENABLE = yes
```

File: keyboards/kradoindustries/krado66/info.json [hunk 18]
```diff
@@ -0,0 +1,118 @@
...
+    "manufacturer": "Krado Industries",
+    "url": "https://www.kradoindustries.com/", 
+    "maintainer": "Krado Industries", 
+    "usb": {
+        "vid": "0x12D0",
+        "pid": "0x1166",
+        "device_version": "2.0.0"
+    },
+    "processor": "atmega32u4",
+    "bootloader": "lufa-ms",
```
Comment (waffle87) [inline keyboards/kradoindustries/krado66/info.json:12]:
```suggestion
    "bootloader": "atmel-dfu",
```

File: keyboards/kradoindustries/kousa/keymaps/default/keymap.c [hunk 19]
```diff
@@ -0,0 +1,127 @@
...
+  }
+#endif
```
Comment (drashna) [inline keyboards/kradoindustries/kousa/keymaps/default/keymap.c:127]: this should be moved to the keyboard's c file.

File: keyboards/kradoindustries/promenade/config.h [hunk 20]
```diff
@@ -0,0 +1,83 @@
...
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
```
Comment (waffle87) [inline keyboards/kradoindustries/promenade/config.h:19]: This file should be removed.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/kradoindustries/promenade/info.json [hunk 21]
```diff
@@ -0,0 +1,109 @@
...
+    "bootloader": "atmel-dfu",
+    "features": {
+        "bootmagic": true,
+        "mousekey": true,
+        "extrakey": true,
+        "console": false,
+        "command": false,
+        "nkro": true
+    },
+
```
Comment (waffle87) [inline keyboards/kradoindustries/promenade/info.json:22]:
```suggestion
        "nkro": true,
        "rgblight": true
    },
    "rgblight": {
        "pin": "B7",
        "led_count": 14,
        "sleep": true,
        "animations": {
            "alternating": true,
            "breathing": true,
            "christmas": true,
            "knight": true,
            "rainbow_mood": true,
            "rainbow_swirl": true,
            "rgb_test": true,
            "snake": true,
            "static_gradient": true,
            "twinkle": true
        }
    },
```

File: keyboards/kradoindustries/promenade/info.json [hunk 22]
```diff
@@ -0,0 +1,109 @@
...
+                
+            ]
+        }
+    }
+}
+
+
+// Replace label with matrix formula 
+// Find "label"\:"(\d+),(\d+)"
+// Replace "matrix": \[\1, \2\]
```
Comment (waffle87) [inline keyboards/kradoindustries/promenade/info.json:109]:
```suggestion
```

File: keyboards/kradoindustries/promenade/keymaps/default/readme.md [hunk 23]
```diff
@@ -0,0 +1 @@
+# The default keymap for Promenade
```
Comment (waffle87) [inline keyboards/kradoindustries/promenade/keymaps/default/readme.md:1]: This file should be removed.

File: keyboards/kradoindustries/promenade/keymaps/via/readme.md [hunk 24]
```diff
@@ -0,0 +1 @@
+# The via keymap for Promenade
```
Comment (waffle87) [inline keyboards/kradoindustries/promenade/keymaps/via/readme.md:1]: This file should be removed.

File: keyboards/kradoindustries/promenade/promenade.h [hunk 25]
```diff
@@ -0,0 +1,35 @@
...
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "quantum.h"
+
```
Comment (waffle87) [inline keyboards/kradoindustries/promenade/promenade.h:21]: This file should be removed.

File: keyboards/kradoindustries/promenade/rules.mk [hunk 26]
```diff
@@ -0,0 +1,14 @@
...
+#MOUSEKEY_ENABLE = no        # Mouse keys
+#EXTRAKEY_ENABLE = yes       # Audio control and System control
+#CONSOLE_ENABLE = no         # Console for debug
+#COMMAND_ENABLE = yes        # Commands for debug and configuration
+#NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+AUDIO_ENABLE = no           # Audio output
+UNICODE_ENABLE = yes        # Unicode
+RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight.  Do not enable this with audio at the same time.
+SLEEP_LED_ENABLE = no  		# Breathing sleep LED during USB suspend
```
Comment (waffle87) [inline keyboards/kradoindustries/promenade/rules.mk:14]:
```suggestion
# This file intentionally left blank
```

**Review by waffle87 — CHANGES_REQUESTED** [review]
Changes outside of the `keyboards` directory need to be reverted. (ie. submodule changes, `.vscode/settings.json`, & `c_cpp_properties.json`)

File: keyboards/kradoindustries/kousa/keymaps/via/readme.md (hunk 7 above)
Comment (Feags) [reply keyboards/kradoindustries/kousa/keymaps/via/readme.md:1]: Removed

File: .vscode/settings.json (hunk 12 above)
Comment (Feags) [reply .vscode/settings.json:30]: Not sure how I changed this. I don't remember opening the file. I think it has been reverted now.

File: c_cpp_properties.json
Comment (Feags) [reply c_cpp_properties.json:1]: Removed

File: keyboards/kradoindustries/kousa/info.json [hunk 27]
```diff
@@ -0,0 +1,156 @@
...
+        "rotary": [
+            { "pin_a": "GP10", "pin_b": "GP9" }
+        ]
+    },
+    "rgblight": {
+        "led_count": 18,
+        "pin": "GP4",
+        "hue_steps": 10,
+        "saturation_steps": 17,
+        "brightness_steps": 17,
```
Comment (zvecr) [inline keyboards/kradoindustries/kousa/info.json:31]: These are defaults, and a value so close to default that it really doesnt offer much value
```suggestion
```

File: keyboards/kradoindustries/kousa/rules.mk [hunk 28]
```diff
@@ -0,0 +1,6 @@
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+ENCODER_ENABLE = yes
+
```
Comment (zvecr) [inline keyboards/kradoindustries/kousa/rules.mk:5]: Shouldnt really mix location of feature enables
```suggestion
```

File: keyboards/kradoindustries/kousa/info.json [hunk 29]
```diff
@@ -0,0 +1,156 @@
...
+        "pid": "0x4B4F",
+        "device_version": "1.0.0"
+    },
+    "features": {
+        "bootmagic": true,
+        "mousekey": true,
+        "extrakey": true,
+        "console": false,
+        "command": false,
+        "nkro": true
```
Comment (zvecr) [inline keyboards/kradoindustries/kousa/info.json:19]:
```suggestion
        "nkro": true,
        "encoder": true,
        "rgblight": true
```

File: keyboards/kradoindustries/krado66/info.json (hunk 15 above)
Comment (zvecr) [reply keyboards/kradoindustries/krado66/info.json:133]: This still needs to be done

File: keyboards/kradoindustries/krado66/rules.mk [hunk 30]
```diff
@@ -0,0 +1,2 @@
+ENCODER_ENABLE = yes
+RGBLIGHT_ENABLE = yes
```
Comment (zvecr) [inline keyboards/kradoindustries/krado66/rules.mk:2]:
```suggestion
# This file intentionally left blank
```

File: keyboards/kradoindustries/krado66/info.json [hunk 31]
```diff
@@ -0,0 +1,133 @@
...
+    },
+    "processor": "atmega32u4",
+    "bootloader": "atmel-dfu",
+    "features": {
+        "bootmagic": true,
+        "mousekey": true,
+        "extrakey": true,
+        "console": false,
+        "command": false,
+        "nkro": true
```
Comment (zvecr) [inline keyboards/kradoindustries/krado66/info.json:19]:
```suggestion
        "nkro": true,
        "encoder": true,
        "rgblight": true
```

File: keyboards/kradoindustries/krado66/info.json [hunk 32]
```diff
@@ -0,0 +1,133 @@
...
+    },
+
+    "rgblight": {
+        "led_count": 14,
+        "pin": "B2",
+        "hue_steps": 12,
+        "saturation_steps": 15,
+        "brightness_steps": 12,
```
Comment (zvecr) [inline keyboards/kradoindustries/krado66/info.json:27]:
```suggestion
```

File: keyboards/kradoindustries/krado66/keymaps/default/keymap.c [hunk 33]
```diff
@@ -0,0 +1,171 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "krado66.h"
```
Comment (zvecr) [inline keyboards/kradoindustries/krado66/keymaps/default/keymap.c:18]:
```suggestion
#include QMK_KEYBOARD_H
```

**Review by zvecr — CHANGES_REQUESTED** [review]
Submodule changes still need to be reverted.

**Comment by drashna** [comment]
It looks like you have changes to the submodules here.  These need to be reverted before the PR can be approved. 

Rebasing sometimes will fix this. Otherwise, you'd want to do something like this to fix them:
```sh
git checkout master
git submodule foreach 'git log -n1 --oneline'
# remember the first abcdef1234 on the submodules that have an issue
git checkout @@your-branch@@
# for each of the problematic repo's:
cd lib/@@name@@
git checkout abcdef1234
```

**Comment by Feags** [comment]
I am unable to revert the submodules. I don't know how or why they are modified at all and the instructions that [drashna](https://github.com/drashna) kind provided have not worked. I am deleting my qmk repos and starting from scratch.

**Pull request closed without merging**